### ISSUE TITLE
Gesture Improvements

### ISSUE THREAD

**Issue by leolost2605** [elementary/gala#2157]
This is a meta issue to track improvements to the gesture system. (Because I'm waiting on reviews and starting to lose track of when I can work on what. I think having it public also won't hurt so feedback or more linked issues is welcome :))

- [x] #2074 

Docks related:
- [ ] #2150 which then allows for
  - [ ] When switching from a workspace where an autohide panel should hide to a workspace where it should show reveal it in sync with the switch animation (I think that's something we want?)

Better Touchscreen gestures:
- [ ] Requires #2096 
  - [ ] #1959 also requires #2150

Misc:
- [ ] Interruptable animations  #1202
- [ ] Touchpad gesture backend in gala (drop touchegg)
- [ ] #1697

Bug Fixes:
- [ ] #2149 
- [ ] #2148

**Comment by lenemter** [comment]
Since we're rewriting our gestures system it would be really nice to consider allowing 'gesture progress' be more than 100 (https://github.com/elementary/gala/issues/1057) or be negative (https://github.com/elementary/gala/issues/1140). This would allow us to have more responsive gestures. But I'm not sure how this can be implemented.

**Comment by leolost2605** [comment]
Yeah that's a good idea I didn't even think about. I'm planning to split out some commits regarding giving more info to gestures from #2096. I think I can add some stuff there that would allow us to implement this pretty easily.
